File: src/product/application/dto/ProductSummary.dto.ts
type ProductSummaryDTO = {
  id: number;
  name: string;
  slug: string;
  short_description: string;
  base_price: number;
  sale_price: number;
  currency: string;
  primary_image: { url: string; alt_text: string };
  brand: { id: number; name: string };
  seller: { id: number; name: string };
  status: string;
  in_stock: boolean;
  rating: number;
  review_count: number;
  created_at: Date;
  updated_at?: Date;
};

export default ProductSummaryDTO;

File: src/main/java/com/wanted_preonboarding_challenge_backend/eCommerce/dto/ProductSummaryDto.java
package com.wanted_preonboarding_challenge_backend.eCommerce.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.wanted_preonboarding_challenge_backend.eCommerce.dto.Image.ImageDto;
import com.wanted_preonboarding_challenge_backend.eCommerce.dto.brand.BrandDto;
import com.wanted_preonboarding_challenge_backend.eCommerce.dto.seller.SellerDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ProductSummaryDto {
    private Long id;
    private String name;
    private String slug;
    @JsonProperty("short_description")
    private String shortDescription;

    @JsonProperty("base_price")
    private Integer basePrice;

    @JsonProperty("sale_price")
    private Integer salePrice;

    private String currency;

    @JsonProperty("primary_image")
    private ImageDto primaryImage;

    private BrandDto brand;
    private SellerDto seller;

    private Double rating;

    @JsonProperty("review_count")
    private Integer reviewCount;

    @JsonProperty("in_stock")
    private Boolean inStock;

    private String status;

    @JsonProperty("created_at")
    private LocalDateTime createdAt;
}
